src/workflows/server/server.py

```diff
@@ -522,6 +522,9 @@ async def _run_workflow(self, request: Request) -> JSONResponse:
                     await wrapper.task
                 except Exception:
                     pass
+            # explicitly close handlers from this synchronous api so they don't linger with events
+            # that no-one is listening for
+            await self._close_handler(wrapper)
```

Comment:
Also adding behavior for the plain run workflow api to clean itself out of memory upon completion